### PR TITLE
[Merged by Bors] - Allow nodes to be started ready for remote smeshing

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
This changes the node startup in a way to allow a node to receive a remote PoST service connection.

## Changes
- if `--smeshing-start` is provided, but no `--smeshing-coinbase` the node will panic during startup
  - if both are provided node operates in supervised mode (smapp usecase)
  - if the provided coinbase is not valid the node will also panic during startup
- if `--smeshing-coinbase` is provided, but not `--smeshing-start` the node will be prepared to accept connections from remote PoST services
  - this additionally requires the following config parameters to be set to actually work (**TODO**: verify those during startup?)
    - `grpc-tls-listener`
    - `grpc-tls-ca-cert`
    - `grpc-tls-cert`
    - `grpc-tls-key`
- if neither `--smeshing-start` nor `--smeshing-coinbase` are provided the node can still be instructed to start (and stop) the supervised PoST mode via GRPC (**required for system tests** - coinbase is passed as argument and verified by node)
- if `--smeshing-coinbase` is not defined, but the TLS listener is set up correctly, remote connections should succeed but the node will not start smeshing (i.e. register at PoET, generate PoST proofs, publish ATXs, etc.)

## Test Plan
TODO: add system tests with remote post service

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Comment by fasmat** [comment]
bors try

File: node/node.go [hunk 1]
```diff
@@ -866,38 +866,34 @@ func (app *App) initServices(ctx context.Context) error {
...
+	}
+	app.Config.POSTService.NodeAddress = u.String()
+	postSetupMgr, err := activation.NewPostSetupManager(
+		app.edSgn.NodeID(),
+		app.Config.POST,
+		app.addLogger(PostLogger, lg).Zap(),
+		app.cachedDB, goldenATXID,
+	)
+	if err != nil {
+		app.log.Panic("failed to create post setup manager: %v", err)
```
Comment (dshulyak) [inline node/node.go]: i think if you return an error it will be logged as fatal

**Review by dshulyak — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**